app/assets/javascripts/reports.js.coffee

```diff
@@ -3,30 +3,46 @@ ready = ->
     $('#report-selector').change ->
       $('#submit').remove()
       $('#genre-selector').remove()
-      if $('#report-selector').val() == "book-popularity"
-        $.ajax
-          type: "get",
-          url: '/genres.json'
-          success: (data, status, response) ->
-            selector = "<fieldset  id='genre-selector'>
-                          <label>Select a Category</label>
-                          <select name='genre' required='required'>
-                            <option></option>
-                            <option value='all'>All Categories</option>"
-            for genre in data
-              selector += "<option value='" + genre.id + "''>" + genre.name + "</option>"
-            selector +=  "</select></fieldset>"
-            $('#genre-selector').remove()
-            $('#reports-form').append(selector).trigger('create')
-          error: ->
-           # TODO: how do show user error?
-          dataType: "json"
+      if $('#report-selector').val() == "book-popularity" || $('#report-selector').val() == "unpopular-books"
+        appendDateRanges()
+      fetchAndAppendGenres()
 
       $(document).on 'change', '#genre-selector', ->
         $('#submit').remove()
-        submit = "<button id='submit'>Submit</button>"
+        submit = "<button id='submit' class='go-btn right'>Submit</button>"
         $('#reports-form').append(submit).trigger('create')
         return
 
+fetchAndAppendGenres=()->
+  $.ajax
+    type: "get",
+    url: '/genres.json'
+    success: (data, status, response) ->
+      selector = "<fieldset  id='genre-selector'>
+                    <label>Select a Category</label>
+                    <select name='genre' required='required'>
+                      <option></option>
+                      <option value='all'>All Categories</option>"
+      for genre in data
+        selector += "<option value='" + genre.id + "''>" + genre.name + "</option>"
+      selector +=  "</select></fieldset>"
+      $('#genre-selector').remove()
+      $('#reports-form').append(selector).trigger('create')
+    error: ->
+     # TODO: how do show user error?
+    dataType: "json"
+
+appendDateRanges=()->
+  startDate = "<fieldset  id='date-range-selector'>
+                <label>Select a Date Range</label>
+                <select name='dates'>
+                  <option value=''>Forever</option>
+                  <option value='last-week'>Last Week</option>
+                  <option value='last-month'>Last Month</option>
+                  <option value='last-year'>Last Year</option>
+                </select>
+              </fieldset>"
+  $('#reports-form').append(startDate).trigger('create')
+
```

Comment:
the appendDateRanges() method will occasionally append a WHOLE butt ton of select boxes to the form. I suggest creating a placeholder div:
```<div id='date-range-selector'></div>```
then calling the replaceWith() method on $('#date-range-selector') instead of using the append() method:
```$('#date-range-selector').replaceWith(startDate).trigger('create')```
<img width="777" alt="screen shot 2015-12-28 at 11 01 03 am" src="https://cloud.githubusercontent.com/assets/2886826/12023918/8b1788ba-ad53-11e5-8efc-b9a4ef685f01.png">